src/client/auth/scram.rs

```diff
@@ -131,10 +134,14 @@ impl ScramVersion {
         &self,
         conn: &mut Connection,
         credential: &Credential,
+        server_api: Option<&ServerApi>,
     ) -> Result<FirstRound> {
         let client_first = self.build_client_first(credential, false)?;
 
-        let command = client_first.to_command(self);
+        let mut command = client_first.to_command(self);
```

Comment:
Similar to my suggestion above, I think it makes sense to modify the methods like `build_client_first` to take in server API as a parameter. The idea here is that all of the logic for building the commands is separate from the authentication flow, which makes each of them easier to understand independent of each other.